black_it/samplers/best_batch.py

```diff
@@ -25,7 +27,41 @@
 
 
 class BestBatchSampler(BaseSampler):
-    """This class implements the best-batch sampler."""
+    """This class implements the best-batch sampler.
+
+    The sampler is a very essential type of genetic algorithm that takes the parameters corresponding
+      to the current lowest loss values and perturbs them slightly in a purely random fashion.
+    The sampler first chooses the total number of coordinates to perturb via a beta-binomial distribution
+      BetaBin(dims, a, b) --where dims is the total number of dimensions in the search space --, it then selects
+      that many coordinate randomly, and perturbs them uniformly within the range specified by 'perturbation_range'.
+
+    """
+
+    def __init__(
+        self,
+        batch_size: int,
+        random_state: Optional[int] = None,
+        max_deduplication_passes: int = 5,
+        a: float = 3.0,
+        b: float = 1.0,
+        perturbation_range: int = 6,
+    ):
+        """
+        Initialize the sampler.
+
+        Args:
+            batch_size: the number of points sampled every time the sampler is called
+            random_state: the random state of the sampler, fixing this number the sampler behaves deterministically
+            max_deduplication_passes: the maximum number of deduplication passes that are made
+            a: the a parameter of the beta-binomial distribution
+            b: the b parameter of the beta-binomial distribution
+            perturbation_range: the range of the perturbation applied. The actual perturbation will be in the range
+                plus/minus the perturbation_range times the precision of the specific parameter coordinate
+        """
+        super().__init__(batch_size, random_state, max_deduplication_passes)
+        self.a = a
+        self.b = b
+        self.perturbation_range = perturbation_range
```

Comment:
It is OK for me. To make the new feature perfect, I would add some value checks on `a`, `b`, and `perturbation_range`. In particular:

- `a > 0.0`
- `b > 0.0`
- `perturbation_range > 1`

Reply:
Thanks, it makes sense. Probablu `perturbation_range` should be greater or equal to 1

Reply:
> Thanks, it makes sense. Probablu `perturbation_range` should be greater or equal to 1

If `perturbation_range = 1`, `integers(1, self.perturbation_range)` gives error:

```
np.random.default_rng().integers(1, 1) # raises ValueError: low >= high
```

Reply:
thanks!